### DISCUSSION TITLE
How to set colored text in Multiline Entry

### DISCUSSION THREAD

**Discussion by sahilpaudel-pe** [fyne-io/fyne#5135] [Q&A]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

I am creating RichText using as below

```
richText := widget.NewMultiLineEntry()
richTextContainer := container.NewScroll(richText)
richTextContainer.SetMinSize(fyne.NewSize(800, 300))
```

I am using it to print my logs in the application ui it can be `error` `warn` `info` and there color associated with it.

### Is it possible to construct a solution with the existing API?

I tried all the resources around the web I could but couldn’t find the solution.

### Describe the solution you'd like to see.

```

richText := widget.NewMultiLineEntry()
richTextContainer := container.NewScroll(richText)
richTextContainer.SetMinSize(fyne.NewSize(800, 300))


// expecting something like
richText.append(someText, "red")
richText.append(someText, "blue")
```

**Comment by dweymouth** [comment]
Please use Discussions for questions instead of Issues next time. I believe you are looking for the ColorName property of the RichTextStyle struct, which can be applied to a `widget.TextSegment` and appended to a RichText: https://docs.fyne.io/api/v2.5/widget/richtextstyle.html

**Reply by sahilpaudel-pe** [reply]
Hey @dweymouth Sure thing will do next time.

Trying out this for the first time tried few things from the doc like below but it didn't work for me.

```

newSegment := &widget.TextSegment{
		Text: message,
		Style: widget.RichTextStyle{
			ColorName: fyne.ThemeColorName("ColorNamePrimary"),
			TextStyle: fyne.TextStyle{Bold: true},
		},
	}

	sc.RichTextContent = append(sc.RichTextContent, newSegment)
	sc.RichText = widget.NewRichText(sc.RichTextContent...)
	sc.RichText.Refresh()

```

**Reply by dweymouth** [reply]
you want `ColorName: theme.ColorNamePrimary`

**Comment by andydotxyz** [comment]
Just for clarity, Entry is not a rich text widget - it is like Label but editable. Maybe you need the proposed RichTextEntry?
